index.js

```diff
@@ -286,6 +286,11 @@ Router.prototype.handle = function handle(req, res, callback) {
         return next(layerError || err)
       }
 
+      // store matched routes
+      if (layer.path) {
+        req.matchedRoutes = (req.matchedRoutes || []).concat(layer.matchedPath.path)
```

Comment:
This is going to create unnecessary arrays.  Can you instead only create a new array the first time, then push to it after?

```javascript
req. matchedRoutes = req.matchedRoutes || []
req.matchedRoutes.push(layer.matchedPath.path)
```

Reply:
Fixed in https://github.com/pillarjs/router/pull/86/files#diff-168726dbe96b3ce427e7fedce31bb0bcR295